gateway/node.go

```diff
@@ -174,22 +182,88 @@ var (
 	_ full.StateModuleAPI = (*Node)(nil)
 )
 
+type Options struct {
+	subHandler             *EthSubHandler
+	lookbackCap            time.Duration
+	stateWaitLookbackLimit abi.ChainEpoch
+	rateLimit              int
+	rateLimitTimeout       time.Duration
+	ethMaxFiltersPerConn   int
+}
+
+type Option func(*Options)
+
+// WithEthSubHandler sets the Ethereum subscription handler for the gateway node. This is used for
+// the RPC reverse handler for EthSubscribe calls.
+func WithEthSubHandler(subHandler *EthSubHandler) Option {
+	return func(opts *Options) {
+		opts.subHandler = subHandler
+	}
+}
+
+// WithLookbackCap sets the maximum lookback duration (time) for state queries.
+func WithLookbackCap(lookbackCap time.Duration) Option {
+	return func(opts *Options) {
+		opts.lookbackCap = lookbackCap
+	}
+}
+
+// WithStateWaitLookbackLimit sets the maximum lookback (epochs) for state queries.
+func WithStateWaitLookbackLimit(stateWaitLookbackLimit abi.ChainEpoch) Option {
+	return func(opts *Options) {
+		opts.stateWaitLookbackLimit = stateWaitLookbackLimit
+	}
+}
+
+// WithRateLimit sets the maximum number of requests per second globally that will be allowed
+// before the gateway starts to rate limit requests.
+func WithRateLimit(rateLimit int) Option {
+	return func(opts *Options) {
+		opts.rateLimit = rateLimit
+	}
+}
+
+// WithRateLimitTimeout sets the timeout for rate limiting requests such that when rate limiting is
+// being applied, if the timeout is reached the request will be allowed.
+func WithRateLimitTimeout(rateLimitTimeout time.Duration) Option {
+	return func(opts *Options) {
+		opts.rateLimitTimeout = rateLimitTimeout
+	}
+}
+
+// WithEthMaxFiltersPerConn sets the maximum number of Ethereum filters and subscriptions that can
+// be maintained per websocket connection.
+func WithEthMaxFiltersPerConn(ethMaxFiltersPerConn int) Option {
+	return func(opts *Options) {
+		opts.ethMaxFiltersPerConn = ethMaxFiltersPerConn
+	}
+}
+
 // NewNode creates a new gateway node.
-func NewNode(api TargetAPI, sHnd *EthSubHandler, lookbackCap time.Duration, stateWaitLookbackLimit abi.ChainEpoch, rateLimit int64, rateLimitTimeout time.Duration) *Node {
-	var limit rate.Limit
-	if rateLimit == 0 {
-		limit = rate.Inf
-	} else {
-		limit = rate.Every(time.Second / time.Duration(rateLimit))
+func NewNode(api TargetAPI, opts ...Option) *Node {
+	options := &Options{
```

Comment:
One problem with this is that the per-host limit needs a new limiter for each host, so you'd need to supply a limiter-factory to to make it consistent. It would be nice to offer the flexibility though because a limiter has more than just one parameter.
Unless you can come up with a way to make the interface consistent between `Node` and `Handler` in terms of supplying limiter params I'll keep it as it is.